code/datums/components/mounted_gun.dm

```diff
@@ -0,0 +1,89 @@
+///This component allows gun mounting on vehicle types
+/datum/component/mounted_weapon
+	///The gun mounted on a vehicle
+	var/obj/item/weapon/gun/mounted_gun
+
+/datum/component/mounted_weapon/Initialize(gun_type)
+	. = ..()
+	if(!istype(parent, /obj/vehicle))
+		return COMPONENT_INCOMPATIBLE
+	if(!(gun_type in subtypesof(/obj/item/weapon/gun)))
+		return COMPONENT_INCOMPATIBLE
+	mounted_gun = new gun_type(parent)
+	//NODROP so that you can't just drop the gun or have someone take it off your hands
+	ADD_TRAIT(mounted_gun, TRAIT_NODROP, MOUNTED_TRAIT)
+
+/datum/component/mounted_weapon/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_BUCKLE, PROC_REF(on_buckle))
+	RegisterSignal(parent, COMSIG_MOVABLE_UNBUCKLE, PROC_REF(on_unbuckle))
+	RegisterSignal(parent, COMSIG_MOUSEDROP_ONTO, PROC_REF(on_mousedrop))
+	RegisterSignal(parent, COMSIG_ATOM_ATTACKBY, PROC_REF(on_attackby))
+	RegisterSignal(parent, COMSIG_ATOM_ATTACK_HAND, PROC_REF(on_attack_hand))
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
+	RegisterSignal(mounted_gun, COMSIG_ITEM_DROPPED, PROC_REF(on_weapon_drop))
+
+/datum/component/mounted_weapon/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_MOVABLE_BUCKLE,
+	COMSIG_MOVABLE_UNBUCKLE,
+	COMSIG_MOUSEDROP_ONTO,
+	COMSIG_ATOM_ATTACKBY,
+	COMSIG_ATOM_EXAMINE,
+	COMSIG_ATOM_ATTACK_HAND,
+	))
+	QDEL_NULL(mounted_gun)
+	return ..()
+
+///Behaviour on buckle. Puts the gun in the buckled mob's hands.
+/datum/component/mounted_weapon/proc/on_buckle(datum/source, mob/living/buckling_mob, force = FALSE, check_loc = TRUE, lying_buckle = FALSE, hands_needed = 0, target_hands_needed = 0, silent)
+	SIGNAL_HANDLER
+	var/obj/vehicle/parent_vehicle = source
+	if(!parent_vehicle.is_driver(buckling_mob))	
```

Comment:
This is still smelly, this is the exact thing to use VEHICLE_CONTROL_EQUIPMENT for.
Just make an is_equipment_controller proc or whatever, and have the wheelchair add the flag in auto_assign_occupant_flags()

Reply:
This is literally not the issue you had with bike sidecar. I don't see the point of this, it'll just re-break it and both the driver and the guy driving get the gun. I really don't get what your problem is.

Reply:
Wdym? I told you not to worry about the sidecar, whoever adds a gun to it has to unshitcode it or whatever. I mean if you really don't want to do it I can merge, but it'll need to get done later anyway.